modules/javafx.graphics/src/main/java/com/sun/glass/ui/mac/MacApplication.java

```diff
@@ -455,4 +460,33 @@ public Map<String, Class<?>> getPlatformKeys() {
             Map.entry("macOS.NSColor.systemYellowColor", Color.class)
         );
     }
+
+    private static final String SUPPRESS_AWT_WARNING_PROPERTY = "javafx.preferences.suppressAppleAwtWarning";
+    private static final String AWT_APPEARANCE_PROPERTY = "apple.awt.application.appearance";
+    private static final String AWT_APPLICATION_CLASS = "NSApplicationAWT";
+    private static final String AWT_SYSTEM_APPEARANCE = "system";
+
+    @SuppressWarnings("removal")
+    private boolean checkSystemAppearance = AccessController.doPrivileged(
+            (PrivilegedAction<Boolean>) () -> !Boolean.getBoolean(SUPPRESS_AWT_WARNING_PROPERTY));
+
+    @Override
+    public void checkPlatformPreferencesSupport() {
+        if (checkSystemAppearance && AWT_APPLICATION_CLASS.equals(applicationClassName)) {
+            checkSystemAppearance = false;
+
+            @SuppressWarnings("removal")
+            String awtAppearanceProperty = AccessController.doPrivileged(
+                (PrivilegedAction<String>) () -> System.getProperty(AWT_APPEARANCE_PROPERTY));
+
+            if (!AWT_SYSTEM_APPEARANCE.equals(awtAppearanceProperty)) {
+                Logging.getJavaFXLogger().warning(String.format(
```

Comment:
Minor: Maybe add newlines to break this message into two or three separate lines?

Reply:
Done. I've also moved the `checkSystemAppearance` assignment out of the if branch (it should be unconditionally set to `false` after the first call).